packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -613,34 +742,53 @@ class MongoshNodeRepl implements EvaluationListener {
     };
   }
 
+  /**
+   * Returns state that is only available after initialize(), and throws
+   * an exception if the shell instance has not been initialized.
+   */
   runtimeState(): MongoshRuntimeState {
     if (this._runtimeState === null) {
-      throw new MongoshInternalError('Mongosh not started yet');
+      throw new MongoshInternalError('Mongosh not initialized yet');
     }
     return this._runtimeState;
   }
 
+  /**
+   * Close all resources held by this shell instance; in particular,
+   * close the ShellInternalState instance and the Node.js REPL
+   * instance, and wait for all output that is currently pending
+   * to be flushed.
+   */
   async close(): Promise<void> {
     const rs = this._runtimeState;
     if (rs) {
       this._runtimeState = null;
       rs.repl.close();
       await rs.internalState.close(true);
-      if (this.output.writableLength > 0) {
-        await once(this.output, 'drain');
-      }
+      await new Promise(resolve => this.output.write('', resolve));
```

Comment:
This is a small drive-by fix – `.writableLength > 0` does not imply that `'drain'` will be emitted.

The new state matches what we do in the logging code as well.